include/boost/geometry/policies/compare.hpp

```diff
@@ -37,92 +38,244 @@ namespace boost { namespace geometry
     on equal x-es then on y, etc.
     If a dimension is specified, only that dimension is considered
 */
+
 template
 <
     typename Point = void,
     int Dimension = -1,
-    typename CSTag = void
+    typename StrategyOrTag = void
+>
+struct less_exact
+{
+    using first_argument_type = Point;
+    using second_argument_type = Point;
+    using result_type = bool;
+
+    inline bool operator()(Point const& left, Point const& right) const
+    {
+        return StrategyOrTag::template compare_type
+            <
+                strategy::compare::less,
+                strategy::compare::equals_exact
+            >::apply(left, right);
+    }
+};
+
+template
+<
+    typename Point = void,
+    int Dimension = -1,
+    typename StrategyOrTag = void
 >
 struct less
 {
-    typedef Point first_argument_type;
-    typedef Point second_argument_type;
-    typedef bool result_type;
+    using first_argument_type = Point;
+    using second_argument_type = Point;
+    using result_type = bool;
+
+    inline bool operator()(Point const& left, Point const& right) const
+    {
+        return StrategyOrTag::template compare_type
+            <
+                strategy::compare::less,
+                strategy::compare::equals_epsilon
+            >::apply(left, right);
+    }
+};
+
+
+template <typename Point, int Dimension>
+struct less<Point, Dimension, void>
+{
+    using first_argument_type = Point;
+    using second_argument_type = Point;
+    using result_type = bool;
 
     inline bool operator()(Point const& left, Point const& right) const
     {
-        typedef typename strategy::compare::services::default_strategy
+        using strategy_type = typename strategy::compare::services::default_strategy
             <
                 strategy::compare::less,
+                strategy::compare::equals_epsilon,
                 Point, Point,
-                Dimension,
-                CSTag, CSTag
-            >::type strategy_type;
+                Dimension
+            >::type;
 
         return strategy_type::apply(left, right);
     }
 };
 
-template <int Dimension, typename CSTag>
-struct less<void, Dimension, CSTag>
+template <int Dimension, typename Strategy>
+struct less<void, Dimension, Strategy>
 {
-    typedef bool result_type;
+    using result_type = bool;
 
     template <typename Point1, typename Point2>
     inline bool operator()(Point1 const& left, Point2 const& right) const
     {
-        typedef typename strategy::compare::services::default_strategy
+        return Strategy::template compare_type
             <
                 strategy::compare::less,
-                Point1, Point2,
+                strategy::compare::equals_epsilon
+            >::apply(left, right);
+    }
+};
+
+// for backward compatibility
+
+template <typename Point, int Dimension>
+struct less<Point, Dimension, boost::geometry::cartesian_tag>
+{
```

Comment:
Ok, I see. Indeed `less` is enhanced now, and it wouldn’t break, and these are the implementations. 
So I’m ok now